libopae/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -847,20 +851,28 @@ fpga_result enum_fpga_metrics(fpga_handle handle)
 		 // DCP RC
 		case FPGA_HW_DCP_RC: {
 
-			result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, FPGA_HW_DCP_RC);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Perforamnce metrics.");
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_fme_perf_path(_token, metrics_path) == FPGA_OK) {
+
+				result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, metrics_path, FPGA_HW_DCP_RC);
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Perforamnce metrics.");
+				}
 			}
 
-			if (_handle->bmc_handle == NULL)
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_bmc_path(_token, metrics_path) == FPGA_OK) {
+
+				if (_handle->bmc_handle == NULL)
 					_handle->bmc_handle = dlopen(BMC_LIB, RTLD_LAZY | RTLD_LOCAL);
 
-			if (_handle->bmc_handle) {
-				result = enum_bmc_metrics_info(_handle,  &(_handle->fpga_enum_metric_vector), &metric_num, FPGA_HW_DCP_RC);
-				if (result != FPGA_OK) {
-					FPGA_MSG("Failed to enumerate BMC metrics.");
-				}
+				if (_handle->bmc_handle) {
+					result = enum_bmc_metrics_info(_handle, &(_handle->fpga_enum_metric_vector), &metric_num, FPGA_HW_DCP_RC);
+					if (result != FPGA_OK) {
+						FPGA_MSG("Failed to enumerate BMC metrics.");
```

Comment:
Seems like an error condition.  Use FPGA_ERR and return an error code?

Reply:
fixed